src/main/java/dev/openfeature/sdk/OpenFeatureClient.java

```diff
@@ -116,11 +116,10 @@ private <T> FlagEvaluationDetails<T> evaluateFlag(FlagValueType type, String key
                     openfeatureApi.getHooks());
 
             hookCtx = HookContext.from(key, type, this.getMetadata(),
-                    provider.getMetadata(), ctx, defaultValue);
+                    provider.getMetadata(), mergeEvaluationContext(ctx), defaultValue);
 
-            EvaluationContext ctxFromHook = hookSupport.beforeHooks(type, hookCtx, mergedHooks, hints);
-
-            EvaluationContext mergedCtx = mergeEvaluationContext(ctxFromHook, ctx);
+            EvaluationContext mergedCtx = hookSupport.beforeHooks(type, hookCtx, mergedHooks, hints);
+            hookCtx.setEvaluationCtx(mergedCtx);    
```

Comment:
Merge the context first, _then_ run the before hooks with that, _then_ update the context passed to the subsequent hooks.